src/mococrw/key.h

```diff
@@ -270,7 +270,55 @@ class AsymmetricKeypair : public AsymmetricPublicKey
      * Loads a private key from an HSM, creating an @ref AsymmetricPublicKey
      * object as a result.
      */
-    static AsymmetricKeypair readPrivateKeyFromHSM(HSM &hsm, const std::string &keyID);
+    static AsymmetricKeypair readPrivateKeyFromHSM(const HSM &hsm, const std::string &keyID);
+
+    /**
+     * @brief Generates RSA keypair on HSM token according to the spec given.
+     * @note libp11 supports fetching the keys only by keyID but generating multiple keys with the
+     * same key is not prohibited by PKCS#11 standard. Care should be taken when generating multiple
+     * keys with the same ID because there are no guarantees the correct one will be fetched given
+     * current libp11 implementation.
+     * @param hsm HSM engine handle
+     * @param spec @ref RSASpec
+     * @param keyID ID of the key on HSM. _Only_ hex values are valid because key IDs are stored as
+     * raw bytes.
```

Comment:
I kind of agree that this then maybe should better be a std::vector<uint8_t>

Reply:
@garw I understand that the HSM expect a certain amount of raw bytes as keyID. But for the user of this interface, I think it would be most convenient to be able to also pass a human readable string as a keyID, as this can be more easily handled / used on caller site.

Why not keep std::string as a data type then, and just change semantics? This data type is perfectly fine for both passing raw bytes, as well as (ASCII / UTF8) strings. So the caller could decide to pass a human readable string, and internally you just check for a maximum size of the data, and treat any bytes as raw data.

Reply:
This important point will be handled in a separate PR. I don't want it to get lost in this already huge PR so stay tuned.

Reply:
All fine, thanks for the consideration.